src/main/java/com/xebia/headerbuddy/models/CertificateDetails.java

```diff
@@ -0,0 +1,54 @@
+package com.xebia.headerbuddy.models;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Date;
+
+public class CertificateDetails {
+    private String type;
+    private int hashCode;
+    private String publicKeyAlgorithm;
+    private String publicKeyFormat;
+    private Date expireDate;
+    private String issuer;
```

Comment:
please make private members final if they are not volatile

Reply:
final variables can only be assigned in the declaration, instance initializer or constructor. We are using a different setter. Reasoning is the way we get the certificate.